cli/cmd/archive.go

```diff
@@ -0,0 +1,95 @@
+package cmd
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/pierrec/lz4/v4"
+)
+
+func decompressTarLz4(ctx context.Context, inputFile string, outputDir string) error {
+	// Open the .tar.lz4 file.
+	file, err := os.Open(inputFile)
+	if err != nil {
+		return errors.Wrap(err, "open file")
+	}
+	defer file.Close()
+
+	// Create an LZ4 reader.
+	lz4Reader := lz4.NewReader(file)
+
+	// Decompress into memory or stream directly.
+	var buf bytes.Buffer
+	if _, err = io.Copy(&buf, lz4Reader); err != nil {
+		return errors.Wrap(err, "decompress lz4")
+	}
+
+	// Open the .tar archive.
+	tarReader := tar.NewReader(&buf)
+
+	// Extract files from the tar archive.
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break // End of archive
+		}
+		if err != nil {
+			return errors.Wrap(err, "read tar archive")
+		}
+
+		// Determine the output path.
+		outputPath := outputDir + "/" + header.Name
```

Comment:
can use `filepath.Join` rather

Reply:
Causes a lint error from gosec:
```
cli/cmd/archive.go:49:17: G305: File traversal when extracting zip/tar archive (gosec)
		outputPath := filepath.Join(outputDir, header.Name)
```
That is why I am not doing it that way.